includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1460,20 +1439,112 @@ protected function build_shipping_methods( $shipping_methods ) {
 		return $shipping;
 	}
 
+
+	/**
+	 * Builds the response object to pass to Payment Request
+	 *
+	 * @since 5.3.0
+	 *
+	 * @param bool   $shipping_pending True is the user hasn't defined yet a shipping address.
+	 * @param string $page             Page where the request was made. It can be `product_page` or `cart_page`.
+	 *
+	 * @return array Data object used to update Payment Request button on the client.
+	 */
+	protected function build_response( $shipping_pending = true, $page = 'product_page' ) {
+		if ( 'product_page' == $page ) {
+			$itemized_display_items = true;
+		} elseif ( 'cart_page' == $page ) {
+			$itemized_display_items = false;
+		}
+
+		$data                 = [];
+		$data['currency']     = strtolower( get_woocommerce_currency() );
+		$data['country_code'] = substr( get_option( 'woocommerce_default_country' ), 0, 2 );
+
+		// Taxes
+		$tax              = WC()->cart->tax_total + WC()->cart->shipping_tax_total;
+		$tax_to_substract = 0;
+
+		if ( wc_tax_enabled() ) {
+			if ( $shipping_pending ) {
+				$tax_to_substract = $tax;
+				$tax              = 0;
+			}
+		}
+
+		// Shipping
+		$shipping              = wc_format_decimal( WC()->cart->get_shipping_total(), WC()->cart->dp );
+		$shipping_to_substract = 0;
+
+		if ( wc_shipping_enabled() && WC()->cart->needs_shipping() ) {
+			$data['requestShipping'] = ( wc_shipping_enabled() && WC()->cart->needs_shipping() );
```

Comment:
Looks like we're repeating the same condition above. I think it's safe to use

```php
$data['requestShipping'] = true;
```

Reply:
nice catch. done 4f4015136823738d5c35b8e0ca146e5e231603c1